File: 1-10/7-reverseInteger/reverseInteger.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        INT_MAX = 2147483647
        if x < 0:
            num = -int(str(-x)[::-1])
            if num >= -INT_MAX:
                return num
            
        else:
            num = int(str(x)[::-1])
            if num <= INT_MAX:
                return num
        
        return 0

File: 1-100/1-10/7-reverseInteger/reverseInteger.cpp
#include <string>
#include <exception>

using namespace std;

class Solution {
public:
    int reverse(int x) {
        string s = to_string(x);
        if (s[0] == '-')
            ::reverse(s.begin()+1, s.end());
        else
            ::reverse(s.begin(), s.end());
        
        try {
            return stoi(s);
        } catch(exception& e) {
            return 0;
        }
    }
};